### PR TITLE
fix(connector): [ACI] Response Handling in case of `ErrorResponse`

### PR DESCRIPTION
**Pull request by Sakilmostak**
## Type of Change

- [x] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

In case of Error Response in ACI, it doesn't contain the `id` field, resulting in deserialization error.

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: crates/router/src/connector/aci.rs [hunk 1]
```diff
@@ -425,10 +425,10 @@ impl
...
         res: types::Response,
     ) -> CustomResult<types::ErrorResponse, errors::ConnectorError> {
-        let response: aci::AciPaymentsResponse =
-            res.response
-                .parse_struct("AciPaymentsResponse")
-                .change_context(errors::ConnectorError::ResponseDeserializationFailed)?;
+        let response: aci::AciErrorResponse = res
+            .response
+            .parse_struct("AciErrorResponse")
+            .change_context(errors::ConnectorError::ResponseDeserializationFailed)?;
```
Comment (ArjunKarthik) [inline crates/router/src/connector/aci.rs:431]: We can move this to common method

**Review by ArjunKarthik — CHANGES_REQUESTED** [review]
Please do the suggested changes

File: crates/router/src/connector/aci.rs (hunk 1 above)
Comment (Sakilmostak) [reply crates/router/src/connector/aci.rs]: Done👍

**Review by ArjunKarthik — APPROVED** [review]

File: crates/router/src/connector/aci.rs [hunk 2]
```diff
@@ -50,6 +50,29 @@ impl ConnectorCommon for Aci {
...
+        Ok(types::ErrorResponse {
+            status_code: res.status_code,
+            code: response.result.code,
+            message: response.result.description,
+            reason: response.result.parameter_errors.and_then(|errors| {
+                errors.first().map(|error_description| {
```
Comment (prasunna09) [inline crates/router/src/connector/aci.rs:67]: here, connector always have only one element or more than one?? if more than one element, can we concatenate all the errors and pass it in reason?

**Review by prasunna09 — CHANGES_REQUESTED** [review]

File: crates/router/src/connector/aci.rs (hunk 2 above)
Comment (Sakilmostak) [reply crates/router/src/connector/aci.rs]: Done👍

File: crates/router/src/connector/aci.rs [hunk 3]
```diff
@@ -50,6 +50,33 @@ impl ConnectorCommon for Aci {
...
+                errors
+                    .into_iter()
+                    .map(|error_description| {
+                        format!(
+                            "Field is {} and the message is {}",
+                            error_description.name, error_description.message
+                        )
+                    })
+                    .collect::<Vec<String>>()
+                    .join("\n")
```
Comment (ArjunKarthik) [inline crates/router/src/connector/aci.rs:76]: lets join this with ;

**Review by ArjunKarthik — CHANGES_REQUESTED** [review]

File: crates/router/src/connector/aci.rs (hunk 3 above)
Comment (Sakilmostak) [reply crates/router/src/connector/aci.rs]: Done👍

**Review by ArjunKarthik — APPROVED** [review]

**Review by prasunna09 — APPROVED** [review]

**Pull request merged**